grpc/src/main/java/com/linecorp/armeria/server/grpc/AbstractUnframedGrpcService.java

```diff
@@ -159,7 +165,8 @@ static void deframeAndRespond(
 
         if (grpcStatus.getCode() != Code.OK) {
             PooledObjects.close(grpcResponse.content());
-            res.complete(unframedGrpcErrorHandler.handle(ctx, grpcStatus, grpcResponse));
+            res.complete(unframedGrpcErrorHandler.handle(ctx, grpcStatus, grpcResponse,
```

Comment:
The current implementation creates a dependency `UnframedGrpcErrorHandler` -> `UnframedGrpcStatusFunction`, both of which are user inputs.
I'm wondering what would be the correct behavior if both custom `UnframedGrpcErrorHandler` and `UnframedGrpcStatusFunction` are defined?

Would the behavior be more well-defined if the two functions are declared sequentially? Let me know what you think
i.e.
```
HttpResponse mappedResponse = unframedGrpcErrorHandler.handle(ctx, grpcStatus, grpcResponse);
if (unframedGrpcStatusMappingFunction != UnframedGrpcStatusMappingFunction.of()) {
    final RequestLogAccess log = ctx.log();
    final Throwable cause;
    if (log.isAvailable(RequestLogProperty.RESPONSE_CAUSE)) {
        cause = log.partial().responseCause();
    } else {
        cause = null;
    }
    final HttpStatus status = unframedGrpcStatusMappingFunction.apply(ctx, grpcStatus, cause);
    mappedResponse = mappedResponse.mapHeaders(
            headers -> headers.toBuilder().status(status).build());
}
```

Reply:
Good point. `UnframedGrpcStatusMappingFunction` and `UnframedGrpcErrorHandler` mutually exclusive. 
How about taking one of them when building a `GrpcService`?
If both `UnframedGrpcStatusMappingFunction` and `UnframedGrpcErrorHandler` are set, we can raise `IllegalStateException`.

Reply:
> How about taking one of them when building a GrpcService?
If both UnframedGrpcStatusMappingFunction and UnframedGrpcErrorHandler are set, we can raise IllegalStateException.

This sounds like a better idea 👍

Reply:
Understood the problem you discussed 🤔 
Let me think more about a better idea too.